charts/agent/templates/configmap.yaml

```diff
@@ -52,12 +52,16 @@ data:
     k8s_coldstart:
       max_parallel_cold_starts: {{ include "agent.parallelStarts" .}}
 {{- end }}
-{{ include "agent.logPriority" . | indent 4 }}
+{{/* check for log level sanity */}}
+{{- if and .Values.logPriority (or (hasKey (default dict .Values.sysdig.settings.log) "console_priority") (hasKey (default dict .Values.sysdig.settings.log) "file_priority")) }}
+  {{- fail "Cannot set logPriority when either sysdig.settings.log.console_priority or sysdig.settings.log.file_priority are set" }}
+{{- end }}
+{{- if .Values.logPriority }}
+  {{- $_ := merge .Values.sysdig.settings (dict "log" (dict "console_priority" .Values.logPriority "file_priority" .Values.logPriority)) }}
```

Comment:
question: so does this mean we would like to have all items in .Values.sysdig.settings to also be in the .Values.logPriority?

Reply:
Not quite. 🙂 What I'm doing here is merging
```
log:
  console_priority: .Values.logPriority
  file_priority: .Values.logPriority
```
with any other values already in `sysdig.settings` by using the merge method (https://helm.sh/docs/chart_template_guide/function_list/#merge-mustmerge)

Reply:
I see, the sysdig.settings takes precedence, so no repetition.

Reply:
Exactly. 🙂 In my previous implementation I tried to guarantee no repetition manually (with the unintended consequence of throwing out any additional log settings 🙁). By using the `merge` function we're letting the language do the hard part for us.

Reply:
thanks helm ❤️